### PR TITLE
Transitioning the documentation from rst to md

### PR DESCRIPTION
**Pull request by alongd**
Adjusted T3's documentation to the new OO version.
Added a tutorial (more should follow) and How-to guides.
Added an auto-deployment GitHub Action (unfortunately, auto-deployment can only be tested once merging)

### CODE REVIEW

**Comment by alongd** [comment]
To compile the documentation locally, update the T3 environment (and activate it), then type `mkdocs serve` in the T3 root directory. Point your browser to [http://localhost:8000](http://localhost:8000) and the documentation should greet you in a new look.
See more [here](https://squidfunk.github.io/mkdocs-material/).

File: docs/how_to.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+## Writing simulation adapters
+
+Coming soon!
+
+
+## Pre-QM, or: T3's iteration 0
+
+Sometimes it is desired to conduct thermodynamic and/or rate coefficient calculations in advance,
+prior to the main T3 iterations. For example, one might want to compute thermodynamic properties
+for all important radicals of a fuel molecule in advance (before RMS generates a model).
```
Comment (kspieks) [inline docs/how_to.md:146]: I think you meant `(before RMG generates a model)`?

File: docs/tutorials/1_no_qm.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+            'model': {'core_tolerance': [0.01, 0.001]}}
+
+t3_object = T3(project='T3_tutorial_1',
+               rmg=rmg_args)
+
+t3_object.execute()
+```
+
+
+Here, the ``rmg_args`` argument defines the database libraries to be used by RMG,
```
Comment (kspieks) [inline docs/tutorials/1_no_qm.md:76]: I really like the highlighted sections in the tutorial!

File: docs/how_to.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+(the "core"). See detailed explanation
+<a href="http://reactionmechanismgenerator.github.io/RMG-Py/users/rmg/input.html#model-tolerances" target="_blank">
+here</a>. The corresponding T3 argument is called `core_tolerance`, and is set under `rmg['model']`.
+This `core_tolerance` argument is a list of floats, each will be used in a respective T3
+iteration. For example, setting `core_tolerance = [0.02, 0.01, 0.005, 0.001]` will make T3 use a
+`toleranceMoveToCore` of `0.02` for running RMG in iteration 1,
+and a `toleranceMoveToCore` of `0.001` for running RMG in iteration 4.
+All iterations further iterations use the last entry in `core_tolerance`. In the above example,
+iterations 5, 5, 7... will use a `toleranceMoveToCore` of `0.001` as well.
+
```
Comment (kspieks) [inline docs/how_to.md:165]: Should this be `iterations 5, 6, 7...` ?

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+<a href="https://www.python.org/" target="_blank">
+    <img src="https://img.shields.io/badge/Python-3.7+-blue.svg" alt="Python">
+</a>
+</p>
+
+---
+
+**Documentation**: <a href="https://reactionmechanismgenerator.github.io/T3/" target="_blank">
+https://reactionmechanismgenerator.github.io/T3/</a>
+
```
Comment (kspieks) [inline docs/index.md:33]: Just to check, I thought this PR was replacing the previous sphinx documentation format. Will this link be updated to point to the new squidfunk format once merged?

File: docs/running.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+The primary arguments specify various options for the different respective packages
+(T3, RMG, and QM which currently only supports
+<a href="https://reactionmechanismgenerator.github.io/ARC/index.html">ARC</a>).
+Of these three, only the ``rmg`` argument is required. The ``qm`` argument must
+be specified if QM-based model refinement is desired
+(in most cases it is!). The ``t3`` argument contains optional T3-related
+directives and should commonly be specified.
+
+The RMG arguments in T3 are written in an underscore_lower_case syntax,
```
Comment (kspieks) [inline docs/running.md:63]: Is it helpful to also refer to this as `underscore_lower_case (snake_case) syntax`?

File: docs/tutorials/1_no_qm.md [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+t3_object = T3(project='T3_tutorial_1',
+               rmg=rmg_args)
+
+t3_object.execute()
+```
+*(This script is complete, it should run "as is")*
+
+
+Next we'll define arguments related to RMG. This argument is a Python dictionary which
+will bw passed to the ``rmg`` argument of the ``T3`` object imported above.
```
Comment (kspieks) [inline docs/tutorials/1_no_qm.md:44]: Should this be `will be passed`?

File: docs/running.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+while many are in a cameCase syntax in RMG.
```
Comment (kspieks) [inline docs/running.md:64]: Should this say `camelCase`?

File: docs/tutorials/1_no_qm.md [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+You can name this argument with any legal Python name, here we use ``rmg_args``.
+This is the only required primary argument -- this simple examples does not
```
Comment (kspieks) [inline docs/tutorials/1_no_qm.md:46]: Should this be `this simple example does not`?

**Comment by alongd** [comment]
Good question regarding the `{.internal-link target=_blank}`: I've seen this format in other repositories that use mkdocs, I hope it'll work correctly (opening the content in a new blank tab) without showing this code to the user. If it won't, we'll remove it. I think we need to see it once it's actually merged...

File: docs/index.md (hunk 4 above)
Comment (alongd) [reply docs/index.md:33]: Yes - Once merged, we'll see whether the auto-deployment thing works as expected (we cannot test it without merging). If it does, it'll auto-push the updated documentation to the T3 `gh-pages` branch. The web page at https://reactionmechanismgenerator.github.io/T3/ takes its contents from that branch. (The auto-deployment should trigger every time a PR is merged (even if it doesn't update the docs), keeping the online docs always up to date.)

**Comment by alongd** [comment]
Thanks for the comments!
I added a fixup, let me know when to squash

File: docs/index.md (hunk 4 above)
Comment (kspieks) [reply docs/index.md:33]: Awesome sounds good

**Comment by kspieks** [comment]
Thanks for the fixing the minor items. I think it looks good to squash and I'll merge soon afterwards so we can see how the `{.internal-link target=_blank}` and auto-deployment work. Thanks for the PR!

**Comment by alongd** [comment]
Squashed!
It may take it a while (~10 min?) after merging to deploy

**Pull request merged**